File: Data Structures & Algorithms/products-of-array-discluding-self/submission-3.cs
public class Solution {
    public int[] ProductExceptSelf(int[] nums) {
       int[] numsPrefix = new int[nums.Length];
       int[] numsPostfix = new int[nums.Length];
       for(int i=0;i<nums.Length;i++){
           if(i==0){
            numsPrefix[i] = 1; 
           }
           else{
            numsPrefix[i] = (numsPrefix[i-1] * nums[i-1]);
           }
       }
    
       for(int i=nums.Length-1;i>=0;i--){
        if(i==nums.Length-1){
            numsPostfix[i] = 1 ; 
        }
        else{
            numsPostfix[i] = numsPostfix[i+1] * nums[i+1];
        }
       }

       for(int i=0;i<nums.Length;i++){
        nums[i] = numsPostfix[i] * numsPrefix[i];
       }
      return nums;
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-3.ts
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums: number[], target: number): number[] {
      const freqMap = new Map<number, number>();

      for(let i=0;i< nums.length; i++){
        const diff = target - nums[i];
        if(freqMap.has(diff)) return [freqMap.get(diff), i];
        freqMap.set(nums[i],i);
      }
      return [-1,-1];
    }
}
